cmd/nacp/nacp.go

```diff
@@ -44,6 +46,58 @@ var (
 	nomadTimeout = 310 * time.Second
 )
 
+// New function to get client IP
+func getClientIP(r *http.Request) string {
+	// Check X-Forwarded-For header first
+	forwarded := r.Header.Get("X-Forwarded-For")
+	if forwarded != "" {
+		return strings.Split(forwarded, ",")[0]
+	}
+
+	// Fall back to RemoteAddr
+	ip, _, _ := net.SplitHostPort(r.RemoteAddr)
+	return ip
+}
+
+func resolveTokenAccessor(transport http.RoundTripper, nomadAddress *url.URL, token string) (*api.ACLToken, error) {
```

Comment:
I'm actual not too knowledgeable when it comes to http lib of golang, what is it we do with the transport here :)

Reply:
ah this was for the tls part right?

Reply:
Yeap. Since we have a transport coming from the upstream function call, we use the same settings to call the nomad endpoint. To ensure if we have the tls set we use it.